src/promptflow-tools/promptflow/tools/aoai_gpt4v.py

```diff
@@ -94,8 +94,13 @@ def list_deployment_names(
     try:
         credential = _get_credential()
         try:
+            if isinstance(connection, AzureOpenAIConnection):
```

Comment:
add comments ? Do have test case for both branches ?

Reply:
The logic looks weird, when can the `connection` be of type `AzureOpenAIConnection`.